src/lib/utils/cuckoo_hashtable.hpp

```diff
@@ -23,7 +23,8 @@ class HashTable : private Noncopyable {
  public:
   explicit HashTable(size_t input_table_size) : _input_table_size(input_table_size) {
     // prepare internal hash tables and fill with empty elements
-    _hashtables.resize(NUMBER_OF_HASH_FUNCTIONS, std::vector<std::shared_ptr<HashElement>>(input_table_size));
+    // can't use resize because elements are not copyable
+    for (size_t i = 0; i < NUMBER_OF_HASH_FUNCTIONS; ++i) _hashtables.emplace_back(input_table_size);
```

Comment:
I am not really a fan of single-line `for`s, I always expect to find curly braces. We don't have anything about this in the styleguide, but I always like to keep [goto fail](https://en.wikipedia.org/wiki/Unreachable_code#goto_fail_bug) in mind :)